### PR TITLE
doc: reformat and improve overview.rst

### PR DESCRIPTION
**Pull request by Alizter**
cc @christinerose could you take a look. I've reformatted the text to fit in the 80 char col and improved a few details here and there. This subsumes your PR in #5972.

### CODE REVIEW

File: doc/overview.rst [hunk 1]
```diff
@@ -5,115 +5,116 @@ Overview
...
+compilation of your libraries, executables, and documentation to the
+installation, setting up of tests, and setting up development tools such as
+Merlin, etc.
+
+In addition to the normal features expected from an OCaml build system, Dune
+provides a few additional ones that separate it from the crowd:
+
+-  You never need to tell Dune the location of things such as libraries. Dune
+   will discover them automatically. In particular, this means that when you
+   want to re-organize your project, you need nothing other than to rename your
```
Comment (christinerose) [inline doc/overview.rst:28]:
```suggestion
   want to reorganise your project, you need nothing other than to rename your
```
Removed hyphen and changed to British spelling (which we have in our Style Guide, but the Dune docs might be American spelling. Do you know?)

File: doc/overview.rst [hunk 2]
```diff
@@ -5,115 +5,116 @@ Overview
...
+-  Things always work the same whether your dependencies are local or installed
+   on the system. In particular, this means that you can insert the source for a
+   project dependency in your working copy, and Dune will start using it
+   immediately. This makes Dune a great choice for multi-project development.
+
+-  Cross-platform: as long as your code is portable, Dune will be able to
+   cross-compile it. Read more in the :ref:`cross-compilation` section.
+
+-  Release directly from any revision: Dune needs no setup stage. To release
+   your project, simply point to a specific Git tag (named revision. Of course,
```
Comment (christinerose) [inline doc/overview.rst:40]:
```suggestion
   your project, simply point to a specific Git tag (named revision). Of course,
```

File: doc/overview.rst [hunk 3]
```diff
@@ -123,8 +124,8 @@ A typical Dune project will have a ``dune-project`` and one or more
 interesting things are: libraries, executables, tests, documents to install,
 etc.
 
-We recommended organizing your project to have exactly one library
-per directory. You can have several executables in the same directory, as long
-as they share the same build configuration. If you'd like to have multiple
+We recommended organizing your project to have exactly one library per
```
Comment (christinerose) [inline doc/overview.rst:127]:
```suggestion
We recommended organising your project to have exactly one library per
```
Same with the British spelling. (I think in previous docs we did this, so if it should be American, I'll make a note of it.)

**Review by christinerose — APPROVED** [review]
Just one hyphen removed and two words changed to British spelling. There are specific notes about that in the suggestions. Approving so you can merge immediately after you accept/reject these changes. Let me know if the spelling needs to be American in the Dune docs moving forward.

File: doc/overview.rst (hunk 1 above)
Comment (Alizter) [reply doc/overview.rst:28]: I don't know if we take British or American English as convention. cc @rgrinberg do you know?

Comment (rgrinberg) [reply doc/overview.rst]: I'd stay just stick what's already in use in dune. I don't have a preference, but let's keep it consistent.

**Comment by Alizter** [comment]
@rgrinberg Can you merge?

**Pull request merged**